lib/dor/services/registration_service.rb

```diff
@@ -18,23 +59,18 @@ def register_object(params = {})
         raise Dor::ParameterError, "Unknown item type: '#{object_type}'" if item_class.nil?
 
         # content_model = params[:content_model]
-        admin_policy  = params[:admin_policy]
+        # parent        = params[:parent]
         label         = params[:label]
         source_id     = params[:source_id] || {}
         other_ids     = params[:other_ids] || {}
         tags          = params[:tags] || []
-        # parent        = params[:parent]
         collection    = params[:collection]
-        pid = nil
-        if params[:pid]
-          pid = params[:pid]
-          existing_pid = SearchService.query_by_id(pid).first
-          unless existing_pid.nil?
-            raise Dor::DuplicateIdError.new(existing_pid), "An object with the PID #{pid} has already been registered."
-          end
-        else
-          pid = Dor::SuriService.mint_id
-        end
+
+        # Check for sourceId conflict *before* potentially minting PID
+        source_id_string = check_source_id [source_id.keys.first, source_id[source_id.keys.first]].compact.join(':')
```

Comment:
The two calls to `keys.first` were in the existing code here:
https://github.com/sul-dlss/dor-services/blob/develop/lib/dor/services/registration_service.rb#L47

I just relocated it.  And I added one layer of ArgumentError checking here:
https://github.com/sul-dlss/dor-services/blob/registration_service_fixes/lib/dor/services/registration_service.rb#L73

Two calls to `keys` is costly only proportional to the size of the Hash.  In this case the max size of the Hash is now 1, so the cost is negligible.  Doesn't read great, but isn't my focus in this PR.

Ideally, the argument should be a single string since that is all the method ever wants to operate on (but that would be a breaking change, i.e. not in this PR).